Assets/Scripts/Controllers/FurnitureSpriteController.cs

```diff
@@ -115,19 +113,40 @@ public Sprite GetSpriteForFurniture(Furniture furn)
         // Otherwise, the sprite name is more complicated.
         spriteName += "_";
 
-        // Check for neighbours North, East, South, West.
+        // Check for neighbours North, Northeast, East, Southeast, South, Southwest, West, Northwest.
         int x = furn.Tile.X;
         int y = furn.Tile.Y;
 
-        spriteName += GetSuffixForNeighbour(furn, x, y + 1, "N");
-        spriteName += GetSuffixForNeighbour(furn, x + 1, y, "E");
-        spriteName += GetSuffixForNeighbour(furn, x, y - 1, "S");
-        spriteName += GetSuffixForNeighbour(furn, x - 1, y, "W");
+        string suffix = string.Empty;
 
-        // For example, if this object has all four neighbours of
+        suffix += GetSuffixForNeighbour(furn, x, y + 1, "N");
+        suffix += GetSuffixForNeighbour(furn, x + 1, y, "E");
+        suffix += GetSuffixForNeighbour(furn, x, y - 1, "S");
+        suffix += GetSuffixForNeighbour(furn, x - 1, y, "W");
+        if (suffix.Contains("N") && suffix.Contains("E"))
```

Comment:
I tested the code and it works. Here is a gist with the 3 functions: https://gist.github.com/TomMalbran/f43ba15b0b8b29f8af034f600707b378

GetSuffixForDiagonalNeighbour basically does what you did. It first check if there are neighbours at the given coords, and then if there is a neighbour at the given position. If you replace the variables in the code, you get the if that you created.